### PR TITLE
RouteGuideServer: use fun expression body

### PR DESCRIPTION
**Pull request by chalin**

### CODE REVIEW

File: examples/src/main/kotlin/io/grpc/examples/routeguide/RouteGuideServer.kt [hunk 1]
```diff
@@ -127,8 +125,8 @@ class RouteGuideServer private constructor(
       }.build()
     }
 
-    override fun routeChat(requests: Flow<RouteNote>): Flow<RouteNote> {
-      return flow {
+    override fun routeChat(requests: Flow<RouteNote>): Flow<RouteNote> =
+      flow {
         // could use transform, but it's currently experimental
```
Comment (chalin) [inline examples/src/main/kotlin/io/grpc/examples/routeguide/RouteGuideServer.kt]: @lowasser - do you find that this comment adds value for the tutorial readers? (If not, shall we drop it?)

Comment (chalin) [reply examples/src/main/kotlin/io/grpc/examples/routeguide/RouteGuideServer.kt]: (If you'd like to keep the comment in the code, I can drop it from the tutorial code excerpt only.)

Comment (lowasser) [reply examples/src/main/kotlin/io/grpc/examples/routeguide/RouteGuideServer.kt]: Let's drop it.

File: examples/src/main/kotlin/io/grpc/examples/routeguide/RouteGuideServer.kt [hunk 2]
```diff
@@ -127,8 +125,8 @@ class RouteGuideServer private constructor(
       }.build()
     }
 
-    override fun routeChat(requests: Flow<RouteNote>): Flow<RouteNote> {
-      return flow {
+    override fun routeChat(requests: Flow<RouteNote>): Flow<RouteNote> =
```
Comment (chalin) [inline examples/src/main/kotlin/io/grpc/examples/routeguide/RouteGuideServer.kt]: @lowasser - it is my impression that this implementation of `routeChat()` actually processes all client request messages before issuing a stream of responses. Is that correct, or am I misunderstanding the code?

File: examples/src/main/kotlin/io/grpc/examples/routeguide/RouteGuideServer.kt (hunk 1 above)
Comment (chalin) [reply examples/src/main/kotlin/io/grpc/examples/routeguide/RouteGuideServer.kt]: Done.

File: examples/src/main/kotlin/io/grpc/examples/routeguide/RouteGuideServer.kt (hunk 2 above)
Comment (lowasser) [reply examples/src/main/kotlin/io/grpc/examples/routeguide/RouteGuideServer.kt]: No, that's not correct.  Flows are a little nonintuitive at first, but what's happening is approximately that `collect` is setting up what amounts to a listener that processes requests as they arrive, and `emit` is immediately sending a response back to the client.

**Pull request merged**